### PR TITLE
feat: add token overview component (status-im#13555)

### LINKED ISSUE #13555: UI Component - Token Overview
This issue is to track the creation of the component: **Token Overview**

You can access the link at https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=919%3A27540.

Please have a chat with the design team before starting with the component, and don't forget to add sizing to the issue

### PR DESCRIPTION
**Pull request by J-Son89**
fixes #13555

### Summary

Implemented token overview component. implemented as two separate components, one for token account total and for token price. Can easily be grouped into one component if needed but current use case in wallet designs have these being used independently -> https://www.figma.com/file/Q92CTbwanXZwSWrHWKfJqS/Wallet-for-Mobile?node-id=0%3A1

#### Platforms
- Android
- iOS

### Steps to test
Open app and go to quo2 components in settings menu

status: ready

### CODE REVIEW

File: src/quo2/components/token_overview.cljs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+   [quo.react-native :as rn]
+   [clojure.string :as string]
+   [status-im.utils.currency :as currencies]
+   [status-im.ui.components.icons.icons :as icons]
+   [quo2.components.text :as text]))
+
+(def container-style {:display :flex :width "100%" :padding-left 20 :padding-right 20 :padding-top 12 :padding-bottom 12})
+
+(defn price-direction [price-change increase decrease neutral]
+  (if (> price-change 0) increase (if (< price-change 0) decrease neutral)))
```
Comment (cammellos) [inline src/quo2/components/token_overview.cljs]: Cond is better I think, a bit clearer

```
(cond
    (> price-change 0) increase
    (< price-change 0) decrease
     :else neutral))
```

File: src/quo2/components/token_overview.cljs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+(defn divider [direction]
+  [rn/view {:style  {:height 10
+                     :margin-left 4
+                     :margin-right 4
+                     :width 1
+                     :background-color (price-direction direction colors/success-50-opa-40 colors/danger-50-opa-40 colors/divider-light)}}])
+
+(defn get-direction [percentage-change]
+  (if (= (js/parseInt percentage-change) 0) 0
```
Comment (cammellos) [inline src/quo2/components/token_overview.cljs]:
```
(if (zero? (js/parseInt percentage-change))
     ....
```

File: src/quo2/components/token_overview.cljs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    :price :string
+    :percentage-change :string
+    }"
+  []
+  (fn
+    [{:keys [currency price  percentage-change] :or {currency :usd price "0.00" percentage-change "0.0"}}]
+    (let [direction (get-direction percentage-change)]
+      [rn/view {:style container-style}
+       [text/text  {:number-of-lines 1
+                    :size  :paragraph-2} "Token Price"]
```
Comment (cammellos) [inline src/quo2/components/token_overview.cljs]: use labels for i18n

`(i18n/label :token-price)` and then you can add them to `translations/en.json` (only English is enough)

File: src/quo2/components/token_overview.cljs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+       [text/text  {:style {:margin-top 4}
+                    :weight :semi-bold
+                    :number-of-lines 1
+                    :size  :heading-2} (str (get-in currencies/currencies [currency :symbol]) price)]
+
+       [rn/view {:style {:display :flex :flex-direction :row :margin-top 6 :align-items :center}}
+        (when (not (= direction 0)) [icons/icon (if (>= direction 0) :main-icons2/price-increase12 :main-icons2/price-decrease12)
```
Comment (cammellos) [inline src/quo2/components/token_overview.cljs]: `(zero? direction)`

File: src/quo2/components/token_overview.cljs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                    :size  :paragraph-1} (str "Account " token " Balance")]
+       [rn/view {:style {:display :flex :flex-direction :row :flex 1 :justify-content :space-between}}
+        [text/text  {:number-of-lines 1
+                     :weight :semi-bold
+                     :size  :heading-1} (str (get-in currencies/currencies [currency :symbol]) account-balance)]
+        [rn/image {:source token-img-src
+                   :style {:height 32
+                           :width 32}}]]
+       [rn/view {:style {:display :flex :flex-direction :row :margin-top 6 :align-items :center}}
+        (when (not (= direction 0)) [icons/icon (if (> direction 0) :main-icons2/price-increase12 :main-icons2/price-decrease12)
```
Comment (cammellos) [inline src/quo2/components/token_overview.cljs]: `(zero? direction)`
`(pos? direction)`

File: src/quo2/screens/token_overview.cljs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+(defn cool-preview []
+  (let [state (reagent/atom {:token  "ETH" :account-balance "3.00" :price "1.00" :percentage-change "-3.0" :currency :usd})]
+    (fn []
+      [rn/view {:margin-bottom 50 :padding 16}
+       [preview/customizer state descriptor]
+       [rn/view {:border :black
+                 :border-width 1
+                 :align-items      :center}
+        [quo2/token-balance (merge @state {:token-img-src (if (= (:token @state) "ETH") eth-token snt-token)})]
```
Comment (cammellos) [inline src/quo2/screens/token_overview.cljs]: `merge` is used mostly when you have a map as the second parameter, not a finite set of known keys.
In that case you can use `assoc`.
In your case:

`(assoc @state :token-img-src (if ....))`
But it's multivariadic, so you can also use it to `assoc` multiple keys, for example

`(assoc some-map :a 1 :b 2)`, equivalent to `(merge some-map {:a 1 :b 2})`, but clearer and probably a tad faster.

File: src/quo2/screens/token_overview.cljs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        [rn/view {:padding-vertical 25
+                  :align-items      :center}]
+        [quo2/token-price (merge @state {:token-img-src (if (= (:token @state) "ETH") eth-token snt-token)})]]])))
```
Comment (cammellos) [inline src/quo2/screens/token_overview.cljs]: Same as above

**Review by cammellos — COMMENTED** [review]
Looks good, just some style comments

File: src/quo2/components/token_overview.cljs (hunk 1 above)
Comment (J-Son89) [reply src/quo2/components/token_overview.cljs]: oh yeah 😅 . thanks! :)

File: src/quo2/components/token_overview.cljs (hunk 3 above)
Comment (J-Son89) [reply src/quo2/components/token_overview.cljs]: Cool, sounds good :)

File: src/quo2/screens/token_overview.cljs (hunk 6 above)
Comment (J-Son89) [reply src/quo2/screens/token_overview.cljs]: thanks for the help :)

**Review by cammellos — APPROVED** [review]

**Pull request merged**